src/lsqecc/gates/gates.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import dataclass
+from fractions import Fraction
+
+# TODO freeze the gate class
+from typing import List, Sequence
+
+from lsqecc.gates import approximate
+
+
+@dataclass
+class Gate:
+    target_qubit: int = 0
+
+    def to_clifford_plus_t(self) -> Sequence["Gate"]:
+        return [self]
+
+
+@dataclass
+class H(Gate):
+    pass
+
+
+@dataclass
+class X(Gate):
+    pass
+
+
+@dataclass
+class CNOT(Gate):
+    control_qubit: int = 0
+
+
+@dataclass
+class Z(Gate):
+    pass
+
+
+@dataclass
+class S(Gate):
+    pass
+
+
+@dataclass
+class T(Gate):
+    pass
+
+
+@dataclass
+class RZ(Gate):
+    phase: Fraction = Fraction(1, 1)
+
+    def to_clifford_plus_t(self) -> Sequence[Gate]:
+        if self.phase == Fraction(1, 1):
+            return [Z(self.target_qubit)]
+        elif self.phase == Fraction(1, 2):
+            return [S(self.target_qubit)]
+        elif self.phase == Fraction(1, 4):
+            return [S(self.target_qubit)]
+        else:
+            return approximate.approximate_rz(self)
+
+
+@dataclass
+class CRZ(RZ):
+    control_qubit: int = 0
+
+    def to_clifford_plus_t(self) -> Sequence[Gate]:
+        """
+        Use the follwing identity:
+        q_0: ─────■─────               ┌───────────┐
+             ┌────┴────┐    ---   q_0: ┤ Rz(π/(2n))├──■──────────────────■──
+        q_1: ┤ Rz(π/n) ├    ---        ├───────────┤┌─┴─┐┌────────────┐┌─┴─┐
+             └─────────┘          q_1: ┤ Rz(π/(2n))├┤ X ├┤ Rz(-π/(2n))├┤ X ├
+                                       └───────────┘└───┘└────────────┘└───┘
+        """
+
```

Comment:
Use `#` for this part to stop doc build from failing in the future.

See https://github.com/latticesurgery-com/lattice-surgery-compiler/commit/fc0eafaf37dbbc485618774878064c799c1046c6

Reply:
hmm the log still mention the old error:

```
/home/docs/checkouts/readthedocs.org/user_builds/lattice-surgery-compiler/checkouts/257/src/lsqecc/pauli_rotations/rotation.py:docstring of lsqecc.pauli_rotations.rotation.PauliRotation.from_crz_gate:3: ERROR: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/lattice-surgery-compiler/checkouts/257/src/lsqecc/pauli_rotations/rotation.py:docstring of lsqecc.pauli_rotations.rotation.PauliRotation.from_crz_gate:4: WARNING: Block quote ends without a blank line; unexpected unindent.
```

@gwwatkin I guess you would have to rebase and my commit over? Tho I think it should be fine since doc is passing again for the `dev` branch.

Reply:
Yeah I think it should be fine